config/api/routes/auth.php

```diff
@@ -36,24 +36,21 @@
             $password = $this->requestBody('password');
 
             try {
-                if ($user = $this->kirby()->auth()->login($email, $password, $long)) {
-                    return [
-                        'code'   => 200,
-                        'status' => 'ok',
-                        'user'   => $this->resolve($user)->view('auth')->toArray()
-                    ];
-                }
+                $user = $this->kirby()->auth()->login($email, $password, $long);
 
-                throw new NotFoundException(['key' => 'user.undefined']);
+                return [
+                    'code'   => 200,
+                    'status' => 'ok',
+                    'user'   => $this->resolve($user)->view('auth')->toArray()
+                ];
             } catch (Throwable $e) {
                 if ($this->kirby()->option('debug') === true) {
                     throw $e;
                 } else {
-                    // catch any kind of login error
+                    // hide the actual error to avoid leaking security-relevant information
```

Comment:
You basically undid my changes from the last version here. I added the option to get proper info about failed logins when Kirby is in debug mode. There were quite a few people who had issues with failed logins and couldn't debug them properly. I know about the security risk, but I still think we need a way to have proper error handling with useful error messages when you actually want to debug this.

Reply:
No, that feature is still there (see the three lines above). I have only cleaned up the code a bit – it still works exactly the same way.